passes/cmds/add.cc

```diff
@@ -106,6 +142,12 @@ struct AddPass : public Pass {
 		log("selected modules.\n");
 		log("\n");
 		log("\n");
+		log("    add {-assert|-assume|-live|-fair|-cover} [-if <name2>] <name1>\n");
```

Comment:
This description doesn't match what the code does: it says you should use `add -assert -if en foo` but what actually works is `add -assert foo -if en` (which is what we want).

And I'm not sure how the argument parsing ends up there, but if you do attempt to use `add -assert -if en foo` or even just `add -assert -if foo` then no error message is generated but nothing happens either...

Reply:
Oops, good catch.  This is actually a very interesting bug!

With `add -assert -if en foo`, the '-assert' is parsed correctly, but assigns "-if" to `arg_name`.  This never has the chance to generate an error message, because the test `if(!design->selected_whole_module(module->name))` succeeds.

This test succeeds because the extra arguments `en` and `foo` are handled by `Pass::extra_args()` in `kernel/register.cc`.  `en` does not begin with '-', so it's not detected as an erroneous argument, and `select` is its default `true` because the call to `extra_args()` does not include a final `false` parameter.  So it treats `en` and `foo` as arguments to a `select` command and calls `handle_extra_select_args()`.

This is definitely very buggy behavior but is actually outside the scope of this PR... I will create a separate PR to fix the call to `extra_args()`.